web_monitoring/db.py

```diff
@@ -1,47 +1,658 @@
+    Build a Version dict from parameters, performing some validation.
+    """
+    if not isinstance(capture_time, str):
+        capture_time = _tzaware_isoformat(capture_time)
+    if source_metadata is None:
+        source_metadata = {}
+    version = {'page_id': page_id,
+               'uuid': uuid,
+               'capture_time': capture_time,
+               'uri': str(uri),
+               'hash': str(hash),
+               'source_type': str(source_type),
+               'title': str(title),
+               'source_metadata': source_metadata}
+    return version
+
+
+def _build_importable_version(*, page_url, uuid, capture_time, uri, hash,
+                             source_type, title, source_metadata=None):
+    """
+    Build a Version dict from parameters, performing some validation.
+
+    This is different than _build_version because it needs ``page_url`` instead
+    of ``page_id`` of an existing Page.
+    """
+    if not isinstance(capture_time, str):
+        capture_time = _tzaware_isoformat(capture_time)
+    if source_metadata is None:
+        source_metadata = {}
+    version = {'page_url': page_url,
+               'uuid': uuid,
+               'capture_time': capture_time,
+               'uri': str(uri),
+               'hash': str(hash),
+               'source_type': str(source_type),
+               'title': str(title),
+               'source_metadata': source_metadata}
+    return version
+
+class MissingCredentials(RuntimeError):
+    ...
 
 
-def query_import_status(import_id):
+class Client:
     """
-    Check on the status of a bulk import job.
+    Communicate with web-monitoring-db via its REST API.
+
+    This object encapsulates authentication information and provides
+    methods corresponding to the REST API.
+
+    The Client can also be configured via environment variables using the
+    class method :meth:`Client.from_env`.
 
     Parameters
     ----------
-    import_id : integer
+    email : string
+    password : string
+    url : string, optional
+        Default is ``https://api.monitoring.envirodatagov.org``.
     """
-    url = WEB_MONITORING_SHOW_IMPORT_API.format(db_url=settings['db_url'],
-                                                import_id=import_id)
-    return requests.get(url,
-                        auth=(settings['db_email'], settings['db_password']))
+    def __init__(self, email, password, url=DEFAULT_URL):
+        self._auth = (email, password)
+        self._api_url = f'{url}/api/v0'
+
+    @classmethod
+    def from_env(cls):
+        """
+        Instantiate a :class:`Client` by obtaining its authentication info from
+        these environment variables:
+
+            * ``WEB_MONITORING_DB_URL``
+            * ``WEB_MONITORING_DB_EMAIL``
+            * ``WEB_MONITORING_DB_PASSWORD`` (optional -- defaults to
+              ``https://api.monitoring.envirodatagov.org``)
+        """
+        try:
+            url = os.environ['WEB_MONITORING_DB_URL']
```

Comment:
This needs to be `os.environ.get()`, otherwise it’ll raise a `KeyError` even though this is now optional.